Tests/src/PublicAPITests/ComponentTests/GaussianSplatComponentTests.cpp

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../AssetSystemTestHelpers.h"
+#include "../SpaceSystemTestHelpers.h"
+#include "../UserSystemTestHelpers.h"
+#include "Awaitable.h"
+#include "CSP/CSPFoundation.h"
+#include "CSP/Common/Optional.h"
+#include "CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h"
+#include "CSP/Multiplayer/Components/ScriptSpaceComponent.h"
+#include "CSP/Multiplayer/MultiPlayerConnection.h"
+#include "CSP/Multiplayer/Script/EntityScript.h"
+#include "CSP/Multiplayer/SpaceEntity.h"
+#include "CSP/Systems/SystemsManager.h"
+#include "CSP/Systems/Users/UserSystem.h"
+#include "TestHelpers.h"
+
+#include "gtest/gtest.h"
+#include <chrono>
+#include <filesystem>
+#include <thread>
+
+
+using namespace csp::multiplayer;
+using namespace std::chrono_literals;
+
+
+namespace
+{
+
+bool RequestPredicate(const csp::systems::ResultBase& Result)
+{
+	return Result.GetResultCode() != csp::systems::EResultCode::InProgress;
+}
+
+#define RUN_GAUSSIAN_SPLAT_TESTS 0 // LOCAL TEST CODE
+
+#if RUN_ALL_UNIT_TESTS || RUN_GAUSSIAN_SPLAT_TESTS || RUN_GAUSSIAN_SPLAT_TEST
+CSP_PUBLIC_TEST(CSPEngine, GaussianSplatTests, GaussianSplatTest)
+{
+	SetRandSeed();
+
+	auto& SystemsManager = csp::systems::SystemsManager::Get();
+	auto* UserSystem	 = SystemsManager.GetUserSystem();
+	auto* SpaceSystem	 = SystemsManager.GetSpaceSystem();
+	auto* AssetSystem	 = SystemsManager.GetAssetSystem();
+	auto* Connection					= SystemsManager.GetMultiplayerConnection();
```

Comment:
Indentation issue or git?

Reply:
Unsure, but will investigate!

Reply:
Ah this was the CSP clang formatter at work. As designed.